File: internal/gemini/prompt.go
package gemini

const dailySummaryPromptTemplate = `Você é um assistente que recebe as atividades
feitas no GitHub hoje, incluindo ações em repositórios privados. Com base
nelas, gere um resumo detalhado em formato de parágrafo:

REQUISITOS:
- Texto em parágrafo corrido, com pelo menos 100-150 palavras
- Sem emojis e sem hashtags
- Seja específico sobre cada atividade realizada
- Mencione nomes dos repositórios, branches, e detalhes técnicos quando relevante
- Descreva o contexto e propósito das mudanças quando possível
- Use linguagem técnica mas acessível
- Evite frases genéricas como "dia produtivo" ou "muito trabalho"
- Conecte as atividades em uma narrativa coesa sobre o trabalho do dia

Atividades do dia:
%s

Gere um texto detalhado e informativo sobre essas atividades de desenvolvimento.` //nolint:misspell // "informativo" is correct in Portuguese

File: src/gemini/prompt.ts
export const DAILY_SUMMARY_PROMPT = `Você é um assistente técnico que analisa atividades
de desenvolvimento no GitHub. Gere um resumo TÉCNICO e DETALHADO em formato de parágrafo.

REQUISITOS OBRIGATÓRIOS:
- Texto em parágrafo corrido, com pelo menos 150-200 palavras
- SEM emojis e SEM hashtags
- Linguagem técnica profissional
- Mencione SEMPRE: repositórios, branches, número de commits, arquivos modificados
- Descreva O QUE foi implementado/modificado tecnicamente
- Explique o IMPACTO e PROPÓSITO das mudanças no contexto do projeto
- Cite mensagens de commit quando relevantes
- Para refatorações: explique o que está sendo migrado/alterado e por quê
- Para features: descreva a funcionalidade implementada
- Para fixes: explique o problema resolvido
- Conecte commits relacionados em uma narrativa técnica coesa
- EVITE: termos vagos como "atualização", "melhoria", "foco substancial"
- PREFIRA: detalhes concretos sobre código, arquitetura, implementação

ANÁLISE DOS DADOS:
Extraia dos eventos:
- Commits: mensagens, arquivos alterados, propósito
- Pull Requests: título, descrição, mudanças propostas
- Issues: problema descrito, contexto
- Branches: finalidade técnica (feature/, bugfix/, refactor/)

Atividades do dia:
%s

Gere um texto técnico, específico e informativo sobre o trabalho de desenvolvimento realizado.`;

export function buildPrompt(events: string): string {
  return DAILY_SUMMARY_PROMPT.replace("%s", events);
}
